common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -365,7 +365,7 @@ protected boolean runAllTasks(long timeoutNanos) {
             // Check timeout every 64 tasks because nanoTime() is relatively expensive.
             // XXX: Hard-coded value - will make it configurable if it is really a problem.
             if ((runTasks & 0x3F) == 0) {
-                lastExecutionTime = ScheduledFutureTask.nanoTime();
+                lastExecutionTime = System.nanoTime();
                 if (lastExecutionTime >= deadline) {
```

Comment:
@normanmaurer - Why don't we explore making something like we used in https://github.com/netty/netty/pull/4388#discussion_r43048270 more generally available to handle timeouts/expiration?

```java
long start = nanoTime();
long expireTime = // 30 seconds...or w/e you need

while (!isExpired(start, expireTime)) {
...
}

private static boolean isExpired(long start, long expireTime) {
  return nanoTime() - start > expireTime;
}
```